modules/e2ee/Context.spec.js

```diff
@@ -0,0 +1,174 @@
+import { Context } from './Context';
+import { ratchet, importKey } from './crypto-utils';
+
+/*
+function hexdump(buffer) {
+    const a = new Uint8Array(buffer);
+    let s = '';
+
+    for (let i = 0; i < a.byteLength; i++) {
+        s += '0x';
+        s += a[i].toString(16);
+        s += ' ';
+    }
+
+    return s.trim();
+}
+*/
+
+/* TODO: more tests
+ * - delta frames
+ * - frame header is not encrypted
+ * - different sendCounts
+ * - different key length
+ * - ratcheting in decodeFunction
```

Comment:
there is a test for this but there should be a test for "it errors when running out of window" :-)